### PR TITLE
Color everything with pygments

### PR DESCRIPTION
**Pull request by Carreau**
Ok, highly work-in progress, need to cleanup **a lot** of things, 
but it works.

screen shot 2015-11-08 at 20 57 04

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ python:
     - 3.5
     - 3.4
     - 3.3
-    - 2.7
```
Comment (captainsafia) [inline .travis.yml]: Good riddance!

File: IPython/core/interactiveshell.py [hunk 2]
```diff
@@ -643,16 +650,22 @@ def init_environment(self):
     def init_encoding(self):
         # Get system encoding at startup time.  Certain terminals (like Emacs
         # under Win32 have it set to None, and we need to have a known valid
-        # encoding to use in the raw_input() method
+        #mencoding to use in the raw_input() method
```
Comment (captainsafia) [inline IPython/core/interactiveshell.py]: Typo?

File: .travis.yml (hunk 1 above)
Comment (Carreau) [inline .travis.yml]: nah, just for testing, i use `super()` which is not Python2 compatible.

File: IPython/core/interactiveshell.py (hunk 2 above)
Comment (Carreau) [inline IPython/core/interactiveshell.py]: yeah, many typo everywhere, I need to clean and rename all the things.

File: IPython/core/interactiveshell.py [hunk 3]
```diff
@@ -482,7 +487,9 @@ def __init__(self, ipython_dir=None, profile_dir=None,
 
         # This is where traits with a config_key argument are updated
         # from the values on config.
+        self._preinit = True
```
Comment (captainsafia) [inline IPython/core/interactiveshell.py]: Any reason this is set to `True` then `False`? 

Figured it would be faster to ask than track it down myself.

Comment (Carreau) [inline IPython/core/interactiveshell.py]: yep, avoid deadloc durring the call of super() which trigger _style_changed, which need init_syntax_highlighting to have been called, which need super() to have been called, which need init_syntax_highlighting to have been called, which need... super()

**Comment by Carreau** [comment]
@captainsafia I wouldn't review that now, it's far from a bearable state :-P

**Comment by captainsafia** [comment]
giphy-9

**Comment by Carreau** [comment]
rebasing/squashing, last commit was 86dc458492b53b5535462578835f12dca073077c

**Comment by Carreau** [comment]
Lady and gentleman, this now works on Python2 ! (again)

**Comment by Carreau** [comment]
@parleur, I dropped ~15 TODO here and there in the files. Some are easy. If you want to try.

**Comment by minrk** [comment]
Looking forward to this. Let me know if you want a hand rebasing and/or when you think this is ready for more review.

**Comment by Carreau** [comment]
> Looking forward to this. Let me know if you want a hand rebasing and/or when you think this is ready for more review.

Yeah, I think I need to squash and basically redo what @takluyver did for recursion stacks, too painful to rebase.

**Pull request closed without merging**